src/renderer/shared/lib/utils/substrate.ts

```diff
@@ -128,3 +130,35 @@ export const getTypeVersions = (api: ApiPromise, typeName: string): string[] =>
   // @ts-ignore
   return api.createType(typeName).defKeys;
 };
+
+export const getTypeName = (api: ApiPromise, transferType: XcmTransferType, paramName: string): string | undefined => {
+  const { pallet, call } = getPalletAndCallByXcmTransferType(api, transferType);
+
+  const param = api.tx[pallet][call].meta.args.find((n) => n.name.toString() === paramName);
+
+  if (param) {
+    return param.type.toString();
+  }
+};
+
+export const getPalletAndCallByXcmTransferType = (
+  api: ApiPromise,
+  transferType: XcmTransferType,
+): { pallet: XcmPallets; call: string } => {
+  if (transferType === 'xtokens') {
+    return { pallet: 'xTokens', call: 'transferMultiasset' };
+  }
+
+  const pallet = api.tx.xcmPallet ? 'xcmPallet' : 'polkadotXcm';
+
+  if (transferType === 'xcmpallet') {
```

Comment:
`const enum XcmTransferType` would be a nice enhancement, what do you think?
To remove string conditions